.gitignore

```diff
@@ -1,151 +1,24 @@
-# Created by http://www.gitignore.io
```

Comment:
I'd prefer not to remove things from this list just yet. In terms of additions, some seem unnecessary for this project. If they were autogenerated by some tool like gitignore.io that would be ok, but if they're by hand I'd prefer to remove them if they're unused.